crates/sui-types/src/epoch_static_info.rs

```diff
@@ -37,8 +55,161 @@ pub struct EpochValidator {
+            name: metadata.name.clone(),
+            description: metadata.description.clone(),
+            image_url: metadata.image_url.clone(),
+            project_url: metadata.project_url.clone(),
+            net_address: Multiaddr::try_from(metadata.net_address.clone()).unwrap(),
+            p2p_address: Multiaddr::try_from(metadata.p2p_address.clone()).unwrap(),
+            consensus_address: Multiaddr::try_from(metadata.consensus_address.clone()).unwrap(),
+            worker_address: Multiaddr::try_from(metadata.worker_address.clone()).unwrap(),
+            voting_power,
+            stake_amount,
+            gas_price,
+            commission_rate,
+        }
+    }
+}
+
+/// Corresponding type to EpochValidator that's json-serializable for RPC responses.
+#[derive(Debug, Default, Serialize, Deserialize, Clone, Eq, PartialEq, JsonSchema)]
+pub struct SuiEpochValidator {
+    pub sui_address: SuiAddress,
+    pub pubkey_bytes: Vec<u8>,
+    pub network_pubkey_bytes: Vec<u8>,
+    pub worker_pubkey_bytes: Vec<u8>,
+    pub proof_of_possession_bytes: Vec<u8>,
+    pub name: String,
+    pub description: String,
+    pub image_url: String,
+    pub project_url: String,
+    pub net_address: Vec<u8>,
+    pub p2p_address: Vec<u8>,
+    pub consensus_address: Vec<u8>,
+    pub worker_address: Vec<u8>,
+
+    pub voting_power: u64,
+    pub stake_amount: u64,
+
+    pub gas_price: u64,
+    pub commission_rate: u64,
+}
+
+impl From<EpochValidator> for SuiEpochValidator {
+    fn from(v: EpochValidator) -> Self {
+        Self {
+            sui_address: v.sui_address,
+            pubkey_bytes: v.pubkey_bytes.as_bytes().to_vec(),
+            network_pubkey_bytes: v.network_pubkey_bytes.as_bytes().to_vec(),
+            worker_pubkey_bytes: v.worker_pubkey_bytes.as_bytes().to_vec(),
+            proof_of_possession_bytes: v.proof_of_possession_bytes,
+            name: v.name,
+            description: v.description,
+            image_url: v.image_url,
+            project_url: v.project_url,
+            net_address: v.net_address.to_vec(),
+            p2p_address: v.p2p_address.to_vec(),
+            consensus_address: v.consensus_address.to_vec(),
+            worker_address: v.worker_address.to_vec(),
+            voting_power: v.voting_power,
+            stake_amount: v.stake_amount,
+            gas_price: v.gas_price,
+            commission_rate: v.commission_rate,
+        }
+    }
+}
+
+impl From<SuiEpochValidator> for EpochValidator {
+    fn from(e: SuiEpochValidator) -> Self {
+        Self::new(
+            &ValidatorMetadata {
+                sui_address: e.sui_address,
+                pubkey_bytes: e.pubkey_bytes,
+                network_pubkey_bytes: e.network_pubkey_bytes,
+                worker_pubkey_bytes: e.worker_pubkey_bytes,
+                proof_of_possession_bytes: e.proof_of_possession_bytes,
+                name: e.name,
```

Comment:
Are we thinking of making some of the validator metadata (name, description, etc) changeable in the middle of an epoch though? @longbowlu @bmwill

Reply:
yeah technically that stuff could change in the middle of the epoch, but for now maybe we don't want to deal with that? i'll leave that up to @longbowlu since they're implementing the rotation functionality

Reply:
I indeed allow these "non-functional" properties to change amid an epoch in this [PR](https://github.com/MystenLabs/sui/pull/8708). Is there a concern with doing this?

Reply:
Ultimately this will only affect UX so I am not too worried right now, e.g. if an explorer wants to always display the up-to-date name, they would have to fetch the latest system state object instead of the epoch info

Reply:
@lxfind  are you going to cache `EpochStaticInfo` for the entire epoch?

Reply:
Yes. I can also rename this to EpochStartInfo, instead of EpochStaticInfo, if that makes things clearer.

Reply:
another approach is in the rpc layer, we could expose a flag for users to indicate if they want the latest info, if yes, then we return the refreshed data

Reply:
I prefer it to be two separate APIs, because the returned value's data structure will be completely different